harness/determined/core/_train.py

```diff
@@ -67,41 +67,58 @@ def _get_last_validation(self) -> Optional[int]:
         logger.debug(f"_get_last_validation() -> {steps_completed}")
         return steps_completed
 
-    def report_training_metrics(
+    def _report_trial_metrics(
```

Comment:
What if a user wants to report trial metrics to be able to save generic metrics? Shouldn't this be a public function so that one could make generic metrics? Or perhaps we should have a separate function for that?

Reply:
I agree the goal of doing this is to allow this to land before 
https://github.com/determined-ai/determined/pull/7117#issuecomment-1629652354 is resolved which is waiting for review https://github.com/determined-ai/determined/pull/7353 so that it's usable sooner.

once all is done we'd want to make these public but I leave that up to the mlsys team.

Reply:
Quick question, #7117 is this PR, did you mean a different one? Or perhaps I'm misreading, it seems like you were saying it was supposed to be a PR we are also waiting on in addition to #7353

Reply:
> Quick question, https://github.com/determined-ai/determined/pull/7117 is this PR, did you mean a different one?
sry I could have been more clear. 

by 
> goal of doing this is to allow this to land before https://github.com/determined-ai/determined/pull/7117#issuecomment-1629652354 is resolved

I meant to say: before the issue of  'lets not proceed because maybe we want to expose a different name instead of `metric type` to end users ' is resolved